thruster/examples/custom_middleware_with_auth.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use log::info;
+use thruster::{
+    context::typed_hyper_context::TypedHyperContext,
+    errors::{ErrorSet, ThrusterError},
+    hyper_server::HyperServer,
+    m,
+    middleware::cookies::HasCookies,
+    middleware_fn, App, HyperRequest, MiddlewareNext, MiddlewareResult, ThrusterServer,
+};
+use tokio::sync::Mutex;
+
+type Ctx = TypedHyperContext<RequestConfig>;
+
+struct User {
+    first: String,
+    last: String,
+}
+
+#[derive(Default)]
+struct ServerConfig {
+    pool: Arc<Mutex<HashMap<String, User>>>,
```

Comment:
Maybe add a comment explaining that it is a "fake" DB pool, some juniors can be confused here

Reply:
I hear you, I think a better solution is to update `pool` to `fake_db` and move on.